webauthn/types.go

```diff
@@ -101,20 +102,28 @@ func (config *Config) validate() error {
 		return fmt.Errorf(errFmtFieldEmpty, "RPDisplayName")
 	}
 
-	if len(config.RPID) == 0 {
-		return fmt.Errorf(errFmtFieldEmpty, "RPID")
-	}
-
 	var err error
 
-	if _, err = url.Parse(config.RPID); err != nil {
-		return fmt.Errorf(errFmtFieldNotValidURI, "RPID", err)
+	var uri *url.URL
+
+	if len(config.RPID) != 0 {
+		if uri, err = url.Parse(config.RPID); err != nil {
```

Comment:
I am not sure if parsing RPID makes sense? This should be generally be a domain name? So parsing as URL has little sense. I see that it was done like this before.

[Spec here](https://www.w3.org/TR/webauthn-2/#rp-id) mentions:

> Though, the [RP ID](https://www.w3.org/TR/webauthn-2/#rp-id) syntaxes MUST conform to either [valid domain strings](https://url.spec.whatwg.org/#valid-domain-string) or URIs [[RFC3986]](https://www.w3.org/TR/webauthn-2/#biblio-rfc3986) [[URL]](https://www.w3.org/TR/webauthn-2/#biblio-url).

But the later, URIs is only allowed for "Other specifications mimicking the [WebAuthn API](https://www.w3.org/TR/webauthn-2/#web-authentication-api) to enable WebAuthn [public key credentials](https://www.w3.org/TR/webauthn-2/#public-key-credential) on non-Web platforms".

So I am not sure, but there is no requirement that it should not be absolute?